src/tblib/__init__.py

```diff
@@ -50,6 +51,8 @@ def __init__(self, frame):
             if k in ("__file__", "__name__")
         }
         self.f_code = Code(frame.f_code)
+        self.f_lineno = int(frame.f_lineno)
+        self.f_back = Frame(frame.f_back) if frame.f_back is not None else None
```

Comment:
This reintroduces the problems of https://github.com/ionelmc/python-tblib/issues/15 and leads to [`RuntimeError`](https://travis-ci.org/ionelmc/python-tblib/jobs/326523392#L1294) when recursion limits are exceeded. The constructor for `Traceback` deals with this problem manually, this would probably need to do the same.

Reply:
You are right. A mimicked the method from `Traceback` and it did the the errors. Not sure if the semantics a correct though:

```python
class Frame(object):
    def _f_globals(self, frame):
        return {
            k: v
            for k, v in frame.f_globals.items()
            if k in ("__file__", "__name__")
        }

    def __init__(self, frame):
        self.f_globals = self._f_globals(frame)
        self.f_code = Code(frame.f_code)
        self.f_lineno = int(frame.f_lineno)
        self.f_back = None

        frame = frame.f_back
        prev_frame = self
        cls = type(self)

        while frame is not None:
            new_frame = object.__new__(cls)
            new_frame.f_globals = self._f_globals(frame)
            new_frame.f_code = Code(frame.f_code)
            new_frame.f_lineno = int(frame.f_lineno)
            prev_frame.f_back = new_frame
            prev_frame = new_frame
            frame = frame.f_back

    def clear(self):
        # For compatibility with PyPy 3.5;
        # clear() was added to frame in Python 3.4
        # and is called by traceback.clear_frames(), which
        # in turn is called by unittest.TestCase.assertRaises
        pass
```